### PR TITLE
Recommendations Block: Add "Open Links in New Tab" option

### LINKED ISSUE #712: Recommendations Block: Add an "Open links in new tab" setting
In some uses cases, users might want to open related links in a new tab. Consider whether this is a valid scenario and if we should implement it.

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR introduces a new "Open Links in New Tab" option in the Recommendation Block's settings.

## Motivation and Context
Closes #712.

## How Has This Been Tested?
- Manual testing, including downgrade/upgrade path.
- Automated E2E testing included within this PR.

## Screenshots
image

### CODE REVIEW

File: src/blocks/recommendations/components/parsely-recommendations-list-item.tsx [hunk 1]
```diff
@@ -11,15 +11,19 @@ import { Recommendation } from '../models/Recommendation';
 interface ParselyRecommendationsListItemProps {
 	imageAlt: string;
 	imagestyle: string;
+	openlinksinnewtab: boolean;
 	recommendation: Recommendation;
 	showimages: boolean;
 }
 
 const getImageForLink = ( imagestyle: string, imageUrl: string, thumbUrlMedium: string ) => imagestyle === 'original' ? imageUrl : thumbUrlMedium;
+const getLinkTarget = ( openlinksinnewtab: boolean ) => Boolean( openlinksinnewtab ) === true ? '_blank' : '_self';
```
Comment (luiztiago) [inline src/blocks/recommendations/components/parsely-recommendations-list-item.tsx]: What do you think to havesomething like:
```
const getLinkObject = ( openlinksinnewtab: boolean ) => Boolean( openlinksinnewtab ) === true ? { target: '_blank' , rel: 'noopener' } : { target: '_self', rel: '' };
```

**Review by luiztiago — COMMENTED** [review]
Only one comment, great job!

File: src/blocks/recommendations/components/parsely-recommendations-list-item.tsx (hunk 1 above)
Comment (acicovic) [reply src/blocks/recommendations/components/parsely-recommendations-list-item.tsx]: Thank you, this led me towards the right way to do this. In my previous attempt I was returning strings and this wasn't working for me.

**Comment by acicovic** [comment]
Note: I've not included `noreferrer` on purpose, but we can discuss it if anyone thinks that it should be included.

**Review by vaurdan — APPROVED** [review]
LGTM :+1:

**Pull request merged**